pkg/operator/controller/garden/reconciler_reconcile.go

```diff
@@ -253,14 +255,29 @@ func (r *Reconciler) reconcile(
 			Fn:           virtualGardenGardenerResourceManager.Wait,
 			Dependencies: flow.NewTaskIDs(deployVirtualGardenGardenerResourceManager),
 		})
+		deployVirtualGardenGardenerAccess = g.Add(flow.Task{
+			Name:         "Deploying Gardener virtual garden access resources",
+			Fn:           virtualGardenGardenerAccess.Deploy,
+			Dependencies: flow.NewTaskIDs(waitUntilVirtualGardenGardenerResourceManagerIsReady),
+		})
+		renewVirtualClusterAccess = g.Add(flow.Task{
+			Name: "Renewing virtual garden access secrets after creation of new ServiceAccount signing key",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return tokenrequest.RenewAccessSecrets(ctx, r.RuntimeClientSet.Client(), r.GardenNamespace)
+			}).
+				RetryUntilTimeout(5*time.Second, 30*time.Second).
+				DoIf(helper.GetServiceAccountKeyRotationPhase(garden.Status.Credentials) == gardencorev1beta1.RotationPreparing),
+			Dependencies: flow.NewTaskIDs(deployVirtualGardenGardenerAccess),
+		})
 		initializeVirtualClusterClient = g.Add(flow.Task{
 			Name: "Initializing connection to virtual garden cluster",
-			Fn: func(ctx context.Context) error {
+			Fn: flow.TaskFn(func(ctx context.Context) error {
 				var err error
-				virtualClusterClient, err = r.initializeVirtualClusterClient(secretsManager)
+				virtualClusterClient, err = r.initializeVirtualClusterClient(ctx)
 				return err
-			},
-			Dependencies: flow.NewTaskIDs(waitUntilVirtualGardenGardenerResourceManagerIsReady),
+			}).
+				RetryUntilTimeout(5*time.Second, 30*time.Second),
```

Comment:
```suggestion
				RetryUntilTimeout(time.Second, 30*time.Second),
```

to make this faster :)